include/rcutils/error_handling.h

```diff
@@ -223,6 +223,20 @@ rcutils_set_error_state(const char * error_string, const char * file, size_t lin
     } \
   } while (0)
 
+/// Indicate that the function intends to set an error message and return an error value.
+/**
+ * \def RCUTILS_CAN_SET_ERROR_MSG_AND_RETURN_WITH_ERROR_OF
+ * Indicating macro similar to RCUTILS_CAN_RETURN_WITH_ERROR_OF, that also sets an error
+ * message.
+ *
+ * \param error_return_value the value returned as a result of a given error.
+ */
+#define RCUTILS_CAN_SET_ERROR_MSG_AND_RETURN_WITH_ERROR_OF(error_return_value) \
```

Comment:
> RCUTILS_CAN_SET_MSG_AND_RETURN_WITH_ERROR_OF

Sounds good too.

> But also, should this macro take an error msg argument instead of building its own?

I thought the same at first. Then it hit me that there's no reasonable error message a user can set to encompass all possible root causes for a given return code, nor it makes sense to duplicate all error messages set elsewhere in the function to stay consistent. We would end up with a bunch of user-defined dummy strings. That's why I made it generic.

Can you think of a use case for a meaningful error message?

Reply:
That's certainly a good point. The use would just be for debugging purposes. The more unique these messages are the easier it can be to track down where they come from. We can hold off until a strong demand arises.